app/assets/javascripts/discourse/app/lib/utilities.js

```diff
@@ -286,8 +286,13 @@ export function determinePostReplaceSelection({
   } else if (selection.start < needle.end) {
     // Selection starts inside needle...
     if (selection.end <= needle.end) {
-      // ... and ends inside needle.
-      return { start: replacement.end, end: replacement.end };
+      if (selection.start >= needle.start && replacement.end <= needle.end) {
```

Comment:
This is insanely specific but I didn't want to introduce a separate option to this function. Basically to handle re-selecting when doing this:

![Peek 2021-12-09 15-29](https://user-images.githubusercontent.com/920448/145339871-5269811e-38af-47cc-97c7-20e52d9deda2.gif)

Basically selecting the text but stopping at the first character on the top line (and not extending into the space) and pressing shift + tab.

Reply:
Without this change the text gets de-selected.

Reply:
This can be a bit dangerous and further complicates the `determinePostReplaceSelection` function. We're adding one more `if` to this function to handle indentations, but we can break selection after replacements. If in the future, somebody decides to change selection after replacement they can accidentally break selection after indentation.

Looks like after indentation, we just select all the lines that were indented. This is a pretty straightforward algorithm. I'm not very familiar with this part of the code and not sure if it's easy, but it possibly can be implemented in a separate isolated function `determinePostIndentationSelection`. This will be less fragile.

> I didn't want to introduce a separate option to this function.

IMO, it's better to go even further and introduce a new function.

Reply:
This can be a bit dangerous and further complicates the determinePostReplaceSelection function. We're adding one more if to this function to handle indentations, but we can break selection after replacements. If in the future, somebody decides to change selection after replacement they can accidentally break selection after indentation.

Looks like after indentation, we just select all the lines that were indented. This is a pretty straightforward algorithm. I'm not very familiar with this part of the code and not sure if it's easy, but it possibly can be implemented in a separate isolated function determinePostIndentationSelection. This will be less fragile.

I didn't want to introduce a separate option to this function.

IMO, it's better to go even further and introduce a new function.

Reply:
I just didn't want to make a completely new flow after `_replaceText()` because `determinePostReplaceSelection` is called from there. Perhaps you are right though, I will introduce an option to skip that altogether after replaceText and instead just select the indented lines manually in a new function after `_indentSelection`